### PR TITLE
cleanups for some dual-funding related code, II of III

### PR DESCRIPTION
**Pull request by niftynei**
This one's mostly small fixes found while running some tests. Plus we completely rework `check_balances`, the method that needed some help anyway.

Based on #4133, start at 3262c3a

### CODE REVIEW

File: openingd/dualopend.c [hunk 1]
```diff
@@ -352,100 +359,216 @@ static bool find_txout(struct wally_psbt *psbt, const u8 *wscript, u16 *funding_
...
 
-static bool check_balances(struct state *state,
+static bool check_balances(const tal_t *ctx,
+			   struct state *state,
 			   struct wally_psbt *psbt,
-			   bool check_opener_balance,
-			   bool do_full_tx_check,
-			   u32 feerate_per_kw_funding)
+			   u32 feerate_per_kw_funding,
+			   char **err_reason)
```
Comment (rustyrussell) [inline openingd/dualopend.c:369]: Better interface would be to return char *, and NULL on success.

**Review by rustyrussell — APPROVED** [review]
Ack a340c1ef396e8d17a42b09a5dafd81e0c3d7606a

File: openingd/dualopend.c (hunk 1 above)
Comment (niftynei) [reply openingd/dualopend.c]: good point!

**Comment by niftynei** [comment]
Redid interface for `check-balances`; rebased on top of #4133

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/dc9bcfb4263782638ad71e0b0a59f35a78ed1819

**Pull request merged**